nixos/modules/services/monitoring/opentelemetry-collector.nix

```
@@ -48,21 +49,27 @@ in {
       description = "Opentelemetry Collector Service Daemon";
       wantedBy = [ "multi-user.target" ];
 
-      serviceConfig = let
-        conf = if cfg.configFile == null
-               then settingsFormat.generate "config.yaml" cfg.settings
-               else cfg.configFile;
-      in
-      {
-        ExecStart = "${getExe opentelemetry-collector} --config=file:${conf}";
-        DynamicUser = true;
-        Restart = "always";
-        ProtectSystem = "full";
-        DevicePolicy = "closed";
-        NoNewPrivileges = true;
-        WorkingDirectory = "/var/lib/opentelemetry-collector";
-        StateDirectory = "opentelemetry-collector";
-      };
+      serviceConfig =
+        let
+          conf =
+            if cfg.configFile == null
+            then settingsFormat.generate "config.yaml" cfg.settings
+            else cfg.configFile;
+        in
+        {
+          ExecStart = "${getExe opentelemetry-collector} --config=file:${conf}";
+          DynamicUser = true;
+          Restart = "always";
+          ProtectSystem = "full";
+          DevicePolicy = "closed";
+          NoNewPrivileges = true;
+          WorkingDirectory = "/var/lib/opentelemetry-collector";
```

not related to this PR at all but shouldn't that be

```suggestion
          WorkingDirectory = "%S/opentelemetry-collector";
```

instead?

---

It used to be on a separate commit, but I was not sure how to keep it according to the contribution guide. So, I squashed it to a single commit.

---

i'm not sure i follow... it is difficult to spot your change now

i think you should either drop the formatting or put in separate commit - what part of contributing guidelines are you referencing?

---

Now there are 3 commits:
1. Fix formatting
2. Add a supplementary group
3. Avoid fixed path

---

wonderful!